From 93a019da62dca31e3881baa2f6acae9691cf3f53 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Thu, 6 Aug 2020 17:24:54 +0530
Subject: [PATCH 27/27] Add DT node for Media Local Bus (MLB)

No pinmux is needed.

Freescale's driver seems incomplete, I don't see any code supporting
differential (6-pin) mode, which SMARC asks for. Therefore I leave MLB
disabled to save power.

Also fix driver file mode and a useless 'error' message.

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 5 +++++
 drivers/mxc/mlb/mxc_mlb.c       | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index bc86f65eeaef..8dec7e8f88e0 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -308,6 +308,11 @@
 };
 #endif
 
+&mlb {		/* SMARC AFB_DIFF1-3 */
+	/* Driver doesn't seem to support differential (6-pin) mode. */
+	//status = "okay";
+};
+
 #if CONFIG_I2S_AUDIO
 &audmux {	/* SMARC I2S0 (instead of SPI0) */
 	pinctrl-names = "default";
diff --git a/drivers/mxc/mlb/mxc_mlb.c b/drivers/mxc/mlb/mxc_mlb.c
index e97cf8892bb0..9f335e7f0282 100755
--- a/drivers/mxc/mlb/mxc_mlb.c
+++ b/drivers/mxc/mlb/mxc_mlb.c
@@ -2681,7 +2681,7 @@ static int mxc_mlb150_probe(struct platform_device *pdev)
 	drvdata->nvcc = devm_regulator_get(&pdev->dev, "reg_nvcc");
 	if (!IS_ERR(drvdata->nvcc)) {
 		regulator_set_voltage(drvdata->nvcc, 2500000, 2500000);
-		dev_err(&pdev->dev, "enalbe regulator\n");
+		dev_info(&pdev->dev, "enalbe regulator\n");
 		ret = regulator_enable(drvdata->nvcc);
 		if (ret) {
 			dev_err(&pdev->dev, "vdd set voltage error\n");
-- 
2.17.1

